e3sm_diags/driver/aerosol_aeronet_driver.py

```diff
@@ -0,0 +1,119 @@
+from __future__ import print_function
+
+import os
+from typing import TYPE_CHECKING
+
+import numpy as np
+import pandas as pd
+from scipy import interpolate
+
+import e3sm_diags
+from e3sm_diags.driver import utils
+from e3sm_diags.plot.cartopy import aerosol_aeronet_plot
+
+if TYPE_CHECKING:
+    from e3sm_diags.parameter.core_parameter import CoreParameter
+
+from e3sm_diags.logger import custom_logger
+
+logger = custom_logger(__name__)
+
+# This aerosol diagnostics scripts based on AERONET sites data was originally developed by Feng Yan and adated and integrated in e3sm_diags by Jill Zhang.
+# Years include 2006–2015 average climatology for observation according to Feng et al. 2022:doi:10.1002/essoar.10510950.1, and Golaz et al. 2022 E3SMv2 paper.
+
+
+def run_diag(parameter: "CoreParameter") -> "CoreParameter":
+    """Runs the tropical cyclone analysis diagnostic.
+
+    :param parameter: Parameters for the run
+    :type parameter: CoreParameter
+    :raises ValueError: Invalid run type
+    :return: Parameters for the run
+    :rtype: CoreParameter
+    """
+    variables = parameter.variables
+    run_type = parameter.run_type
+    seasons = parameter.seasons
+
+    for season in seasons:
+        test_data = utils.dataset.Dataset(parameter, test=True)
+        parameter.test_name_yrs = utils.general.get_name_and_yrs(
+            parameter, test_data, season
+        )
+        parameter.ref_name_yrs = "AERONET (2006-2015)"
+
+        for var in variables:
+            logger.info("Variable: {}".format(var))
+            parameter.var_id = var
+
+            test = test_data.get_climo_variable(var, season)
+            test_site = interpolate_model_output_to_obs_sites(test, var)
+
+        if run_type == "model_vs_model":
+            ref_data = utils.dataset.Dataset(parameter, ref=True)
+            parameter.ref_name_yrs = utils.general.get_name_and_yrs(
+                parameter, ref_data, season
+            )
+            ref = ref_data.get_climo_variable(var, season)
+            ref_site = interpolate_model_output_to_obs_sites(ref, var)
+
+        elif run_type == "model_vs_obs":
+            ref_site = interpolate_model_output_to_obs_sites(None, var)
+        else:
+            raise ValueError("Invalid run_type={}".format(run_type))
+
+        parameter.output_file = (
+            f"{parameter.ref_name}-{parameter.var_id}-{season}-global"
+        )
+        aerosol_aeronet_plot.plot(test, test_site, ref_site, parameter)
+
+    return parameter
+
+
+def interpolate_model_output_to_obs_sites(var, var_id):
```

Comment:
I tried to do type checking but it seems Union[TransientVariable, type[None]] would trigger a pre-commit hook error around Line 112 for no getLatitude() method for None...So I skipped type checking.